<a href="https://colab.research.google.com/github/deboraluz/deboraluz/blob/main/Profissao_Cientista_de_Dados_M13_Projeto.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **MÓDULO 13**
# Projeto - Fundamentos da Descoberta de Dados

Nesse projeto trabalharemos com a base de dados de produtos de um supermercado do Chile.
A ideia é que vocês apliquem os conceitos estatísticos vistos no último módulo, mais os conceitos de visualizações de dados através de gráficos e finalizem publicando no seu github!

In [7]:
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px

Faça a leitura dos dados do arquivo CSV:


Altere o código abaixo de acordo com seu diretório.

In [16]:
df = pd.read_csv("/content/MODULO7_PROJETOFINAL_BASE_SUPERMERCADO.csv", sep=',')

df.head(10)

,title,Marca,Preco_Normal,Preco_Desconto,Preco_Anterior,Desconto,Categoria
0,"Pack 12 un, Leche extra proteína 1 L",Loncoleche,19788,0,0,0,lacteos
1,"Pack 12 un, Leche chocolate receta original 1 L",Soprole,18228,0,0,0,lacteos
2,"Pack 12 un, Leche semidescremada chocolate 1 L",Soprole,18228,0,0,0,lacteos
3,"Pack 12 un, Leche semidescremada frutilla 1 L",Soprole,18228,0,0,0,lacteos
4,"Pack 12 un, Leche sin lactosa chocolate 1 L",Loncoleche,17988,0,0,0,lacteos
5,"Pack 12 un, Leche sin lactosa frutilla 1 L",Loncoleche,17988,0,0,0,lacteos
6,"Pack 12 un, Leche saborizada light chocolate 1 L",Loncoleche,17988,0,0,0,lacteos
7,"Pack 12 un, Leche saborizada frutilla 1 L",Colun,17388,0,0,0,lacteos
8,"Pack 12 un, Leche saborizada vainilla 1 L",Colun,17388,0,0,0,lacteos
9,"Pack 12 un, Leche saborizada manjar 1 L",Colun,17388,0,0,0,lacteos


Os campos do nosso dataframe são:

**Title:** Nome do produto.


**Marca:** A marca do produto.


**Preco_Normal:** O preço em que o produto costuma ser vendido quando não há desconto.


**Preco_Desconto:** O preço vendido após o desconto ser aplicado.


**Preco_Anterior:** Preço em que era comercializado o produto antes do desconto aplicado.


**Desconto:** Total de desconto aplicado.






As colunas que aparecem com valores 0 são para os produtos onde não tivemos descontos aplicados.


As categorias estão em espanhol!

# 1 - Traga a média e a mediana dos preços - coluna Preco_Normal - por categoria de produto.
# Identifique as categorias que parecem ter um valor de média abaixo ou acima da mediana.

In [27]:
#Media
df.groupby('Categoria')['Preco_Normal'].mean().reset_index().sort_values(by='Preco_Normal', ascending=False)

,Categoria,Preco_Normal
1,comidas-preparadas,3095.043478
5,lacteos,2385.219239
2,congelados,2108.042553
0,belleza-y-cuidado-personal,1783.556485
3,frutas,1724.473684
6,verduras,1343.296875
4,instantaneos-y-sopas,765.491228


In [19]:
#Mediana
df.groupby('Categoria')['Preco_Normal'].median().reset_index().sort_values(by='Preco_Normal', ascending=False)

,Categoria,Preco_Normal
1,comidas-preparadas,3290.0
0,belleza-y-cuidado-personal,1569.0
2,congelados,1519.0
3,frutas,1195.0
6,verduras,1180.0
5,lacteos,989.0
4,instantaneos-y-sopas,439.0


Apenas a categoria "comidas-preparadas" possui a média abaixo da mediana, as demais todas a média está acima da mediana.

# 2 - Traga o desvio padrão por categoria de produto.
# Qual o comportamento da média e mediana nas categorias com maior desvio?

In [25]:
#Desvio Padrão
desvio_padrao_por_categoria = df.groupby('Categoria')['Preco_Normal'].std().reset_index().sort_values(by='Preco_Normal', ascending=False)
desvio_padrao_por_categoria

,Categoria,Preco_Normal
5,lacteos,3925.816164
0,belleza-y-cuidado-personal,2210.041719
2,congelados,2111.539896
1,comidas-preparadas,2019.911428
3,frutas,1639.151114
4,instantaneos-y-sopas,1170.232869
6,verduras,1012.699625


Para as categorias com maior desvio padrão:

1.  **Lácteos (lacteos)**:
    *   **Desvio Padrão:** 3925.82 (o maior)
    *   **Média:** 2385.22
    *   **Mediana:** 989.00
    *   **Comportamento:** A média é significativamente maior que a mediana, indicando uma distribuição assimétrica à direita (skewed right). Isso sugere que existem produtos de 'Lácteos' com preços consideravelmente mais altos que puxam a média para cima, enquanto a maioria dos produtos está concentrada em preços mais baixos.

2.  **Beleza e Cuidados Pessoais (belleza-y-cuidado-personal)**:
    *   **Desvio Padrão:** 2210.04 (o segundo maior)
    *   **Média:** 1783.56
    *   **Mediana:** 1569.00
    *   **Comportamento:** A média é um pouco maior que a mediana, o que também aponta para uma leve assimetria à direita. Há uma variabilidade nos preços, mas menos extrema do que em 'Lácteos'.

3.  **Congelados (congelados)**:
    *   **Desvio Padrão:** 2111.54 (o terceiro maior)
    *   **Média:** 2108.04
    *   **Mediana:** 1519.00
    *   **Comportamento:** A média é visivelmente maior que a mediana, indicando uma distribuição assimétrica à direita. Similar a 'Lácteos', há produtos 'Congelados' com preços mais elevados que aumentam a média, enquanto a maior parte dos produtos tem preços mais baixos.

# 3 - Plot um boxplot da distribuição do Preco_Normal para a categoria que você identificou que tem o maior desvio padrão. Como é a distribuição desses dados segundo o boxplot? Você identifica muitos outliers?

Dica: Para trazer apenas os dados da categoria que você deseja você pode usar o df.loc[df['Categoria'] == 'CATEGORIA ESCOLHIDA'

In [32]:
#BoxPlot
lacteos = df.loc[df['Categoria'] == 'lacteos']

fig = px.box(lacteos,
             y='Preco_Normal',
             color_discrete_sequence=['royalblue'],
             title='Boxplot do Preço Normal - Categoria Lácteos')

fig.update_layout(yaxis_title='Preço Normal',
                  xaxis_title='')
fig.show()

O boxplot da categoria lacteos mostra que a distribuição dos preços é assimétrica à direita. A mediana é de aproximadamente 989, enquanto o Q3 é 2619, e o limite superior é cerca de 5934. Acima desse valor há diversos outliers, indicando a presença de muitos produtos com preços significativamente maiores que a maior parte da categoria.

Esses outliers elevam a média (2385,22) em relação à mediana (989,00) e explicam o fato de a categoria apresentar o maior desvio padrão (3925,82) entre todas as categorias analisadas. Portanto, conclui-se que os preços dos produtos da categoria lacteos possuem alta variabilidade e são fortemente influenciados por produtos de alto valor.

# 4 - Plote um gráfico de barras onde temos a média de descontos por categoria.

In [33]:
#Grafico de Barras
media_desconto = (
    df.groupby('Categoria')['Desconto']
      .mean()
      .reset_index()
      .sort_values(by='Desconto', ascending=False))

fig = px.bar(
    media_desconto,
    x='Categoria',
    y='Desconto',
    title='Média de Descontos por Categoria',
    text_auto='.2f')

fig.update_layout(
    xaxis_title='Categoria',
    yaxis_title='Média de Desconto')

fig.show()

# 5 - Plote um gráfico de mapa interativo agrupando os dados por categoria, marca e trazendo a média de desconto.

In [36]:
#Treemap
media_desconto = (df.groupby(['Categoria', 'Marca'])
      .agg(
          Media_Desconto=('Desconto', 'mean'),
          Quantidade=('Desconto', 'count'))
      .reset_index())

fig = px.treemap(media_desconto,
    path=['Categoria', 'Marca'],
    values='Quantidade',
    color='Media_Desconto',
    color_continuous_scale='Blues',
    title='Média de Desconto por Categoria e Marca')

fig.show()